### ISSUE TITLE
sfm reconstruct not available in python bindings

### ISSUE THREAD

**Issue by miguelriemoliveira** [Itseez/opencv_contrib#636]
Hello 

I am trying to use the sfm module from python3.

I can run the [reconstruct example](http://docs.opencv.org/3.1.0/d5/dab/tutorial_sfm_trajectory_estimation.html#gsc.tab=0) like this

```bash
./example_sfm_trajectory_reconstruction desktop_tracks.txt 1914 640 360
```

and I can compile a cpp file with that code as well after using the tips from #476

However, in python, when I import the cv2 module I cannot find the reconstruct method, e.g.,

```python
import cv2
help(cv2.sfm)
```

displays the following [stuff](https://www.dropbox.com/s/jq9ag0o678mp4rj/sfm_display.txt?dl=0)

Basically it seems all sfm functions are available except the sfm.reconstruct.

This is a very similar issue to #476 but the solution in that case for cpp does not work in python.

Any help would be appreciated.

Miguel

**Comment by miguelriemoliveira** [comment]
I found the answer so I will share it.

The reason is that the definition of the reconstruct function in the modules/sfm/include/opencv2/sfm/reconstruction.hpp file 

is defined with the 

CV_EXPORTS, e.g., 

```cpp
CV_EXPORTS_W
void
reconstruct(InputArrayOfArrays points2d, OutputArray Ps, OutputArray points3d, InputOutputArray K,
bool is_projective = false);
```

and the CV_EXPORTS flag does not export python bindings:

"All external functions and classes must use CV_EXPORTS, otherwise there will be linking errors on Windows. For functions/classes which you want to expose in Python, Java etc. you should use CV_EXPORTS_W instead of CV_EXPORTS, but please be careful with overloaded functions and methods, with non-standard parameter types etc."

from [this](http://code.opencv.org/projects/opencv/wiki/CodingStyleGuide)

So I changed from CV_EXPORTS to CV_EXPORTS_W, recompiled and reinstalled opencv and it works.

I do not know why the author of this code "decided" to not use the python bindings, perhaps because of some problem I will encounter further ahead ...

Could not add CV_EXPORTS_W to the overloaded reconstruct methods which had as arguments
const std::vector<std::string> images

**Comment by johncant** [comment]
@miguelriemoliveira - thankyou for posting those changes. Do you have any working python examples using these functions?

**Comment by miguelriemoliveira** [comment]
Hi @johncant . Sorry, I do not. Best of luck.

**Comment by ddrmorais** [comment]
> @miguelriemoliveira - thankyou for posting those changes. Do you have any working python examples using these functions?

I also need these examples. If you find. Please let me know.
